### ISSUE TITLE
Fully deprecate Wizard pattern - sitewide 508-defect-1 ❗ Launchblocker [KEYBOARD]: Radio options MUST be navigable using the keyboard

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#17766]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a keyboard and screen reader user, I expect to be able to use my `up` and `down` arrow keys to navigate through radio buttons so that I can read all available options.

## Details

Several issues exist on the request an informal conference page due to conditional items expanded based on what radio button is selected including:
- Upon expanding, keyboard focus is lost from the radio group so users are unable to continue exploring other options.
- On Safari VoiceOver, it is not announced to the user that their selection has expanded more information.
- On Firefox/NVDA, focus is lost and jumps to the heading 1 "Request a Higher Level Review" when additional fields expand and collapse instead of being maintained within the radio group.

## Steps to Recreate

This issue exists on the informal conference page and the review page.

## Solution

@luke-gcio @Mottie @kevinstachura 
Let's set up a meeting to talk through this page. It may be more appropriate to split this into two separate pages.

## Screenshots or Trace Logs
https://user-images.githubusercontent.com/14154792/102725913-62101880-42e8-11eb-8679-8423f2eb4026.mov

**Comment by Mottie** [comment]
This requires changes to the platform code... It was a simple change in this PR - https://github.com/department-of-veterans-affairs/vets-website/pull/15443

**Comment by kevinstachura** [comment]
Hey team! Was able to mock these up into separate pages:

https://vsateams.invisionapp.com/share/JHZTPGYBEYV#/441441502_HLR-Request-Informal-Conference-2-Who-To-Call

Let me know if you have any questions/thoughts! Both flows ('yes, call me' and 'yes, call my rep') should be viewable in the Invision proto. 👍

**Comment by joshkimux** [comment]
@kevinstachura Thank you Kevin! This gives us more options to work with to approach this defect. 💯 

To review, we now have two possible options for addressing this ticket:

- Fix radio buttons so they're navigable with a keyboard and add a [fieldset to the "yes call my rep" questions](https://github.com/department-of-veterans-affairs/va.gov-team/issues/16261)
- Split the options into separate pages (we may also have to discuss how this will be handled on the check answers page)

@Mottie I'll look to you to pick the path of least resistance as you know the form system's constraints the best. Do let me know if there's anything I can do to support / make this easier on you 🙇

**Comment by kevinstachura** [comment]
Thanks @joshkimux ! Once @Mottie takes a look at it and we decide on a path forward, I will give the finalized design concepts to Peggy so she can ensure that everything looks good from a content perspective. Thanks team! 💯

**Comment by luke-gcio** [comment]
Comment from Robin: Should be addressed when we split the pages but we really need to have platform look at this as an issue in the form system.

**Comment by meganhkelley** [comment]
Passing off to the Design System team, who owns the forms library now! @caw310 @cvalarida

**Comment by joshkimux** [comment]
Note here: this has been fixed within the HLR flow, so I'll be removing this from the HLR accessibility review epic

**Comment by k80bowman** [comment]
@CourtneyChambers77 could you please check if this is a problem with our [va-radio](https://design.va.gov/storybook/?path=/docs/components-va-radio--default) web component?

**Comment by joshkimux** [comment]
@k80bowman I can chime in here as the person who wrote the ticket. The va-radio web component works golden. 

The issue traces back to the old wizard (which is being deprecated) in the form system that dynamically added content based on radio selections. Upon doing so, it caused those radio selections to lose focus (which ultimately led to the issue).

**Comment by k80bowman** [comment]
Thank you for the clarification @joshkimux. Do you feel like this is an issue we need to continue to look into or should we close this ticket?

**Comment by joshkimux** [comment]
I think this would directly be a part of the work in deprecating the old wizard. 

If the old wizard is successfully deprecated and replaced with a more accessible pattern, this could be closed out without any problems. However, if there are still many instances of the old wizard on VA.gov where this issue will likely still exist, it would be something to prioritize and look into.

**Comment by k80bowman** [comment]
What do you mean by the "old wizard." Are you talking about the current forms system?

**Comment by joshkimux** [comment]
I can clarify! [By the "old wizard"-- I'm referring to this pattern](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/399) which is [now in the process of being deprecated](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/399#issuecomment-990011476)

**Comment by joshkimux** [comment]
@k80bowman [some more context is in this closed PR](https://github.com/department-of-veterans-affairs/vets-website/pull/15443#discussion_r552827222)

> `ExpandingGroup` is the culprit for focus loss, it feels like the solution should be in there, not in the component which uses it (`RadioWidget`)

The group came to the conclusion that it would be better to fix this from a design perspective by splitting items onto separate pages (which is another reason why the wizard as it currently is designed is being deprecated). 

That being said:
- We may want to look into finding where the deprecated wizard still exists on VA.gov (and checking if this issue still exists there)
- This ticket can be closed once the former instance of the wizard no longer exists on va.gov

**Comment by k80bowman** [comment]
This is great, thank you for the additional context @joshkimux!

@CourtneyChambers77 I don't think at this point that we need you to look into this.

**Comment by taylorkaren** [comment]
Remaining problem is related to a wizard that is being deprecated. Forms Library team does not need to do any work on this issue.

**Comment by humancompanion-usds** [comment]
@caw310 - To resolve this issue we need a developer to go through and find instances of the old wizard pattern and then I'd like you to work with that developer to file bugs against the teams using that old pattern asking them to deprecate it. If no team exists, assign it to the PO for that area. Once those issues are filed we can close this issue.

To be clear, it's not on the Design System Team to replace deprecated patterns. Once we make the new pattern available it is the responsibility of VFS teams to replace the deprecated pattern.

**Comment by Mottie** [comment]
I know 526 and HLR are still using the old wizard. We need some design & content work to be done prior to making the change.

**Comment by caw310** [comment]
Created ticket for DST engineer to find instances of old wizard pattern. https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1836

**Comment by caw310** [comment]
Closing as this is old. We will do any work in https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1836
